test/engineSpec.js

```diff
@@ -1081,4 +1155,720 @@ describe('Engine', function() {
+            hmdaJson = JSON.parse(JSON.stringify(require('./testdata/complete.json')));
+            topLevelObj = hmdaJson.hmdaFile.transmittalSheet;
+            global._HMDA_JSON.hmdaFile = hmdaJson.hmdaFile;
+        });
+
+        it('should add properties to the error object', function(done) {
+            var error = {'properties': {}};
+            var retrieveProps = rewiredEngine.__get__('retrieveProps');
+            retrieveProps(error, topLevelObj, ['institutionName', 'respondentZip', 'recordID']);
+
+            var expectedError = {
+                'properties': {
+                    'institutionName': 'MIKES SMALL BANK   XXXXXXXXXXX',
+                    'respondentZip': '99999-9999',
+                    'recordID': '1'
+                }
+            };
+
+            expect(Object.equals(error, expectedError)).to.be(true);
+            done();
+        });
+
+        it('should return an exception for a non-existent property', function(done) {
+            var error = {'properties': {}};
+            var retrieveProps = rewiredEngine.__get__('retrieveProps');
+
+            expect(function() {
+                retrieveProps(error, topLevelObj, ['loanNumber']);
+            }).to.throw('Failed to resolve argument!');
+            done();
+        });
+
+    });
+
+    describe('handleArrayErrors', function() {
+        var hmdaJson = {};
+        var topLevelObj = {};
+
+        beforeEach(function() {
+            hmdaJson = JSON.parse(JSON.stringify(require('./testdata/complete.json')));
+            topLevelObj = hmdaJson.hmdaFile.transmittalSheet;
+            global._HMDA_JSON.hmdaFile = hmdaJson.hmdaFile;
+        });
+
+        it('should return a list of errors for a list of line numbers', function(done) {
+            var handleArrayErrors = rewiredEngine.__get__('handleArrayErrors');
+            var array_errors = require('./testdata/array-errors.json');
+
+            expect(Object.equals(handleArrayErrors(hmdaJson.hmdaFile, [1, 3], ['recordID', 'filler']), array_errors)).to.be(true);
+            done();
+        });
+    });
+
+    describe('handleUniqueLoanNumberErrors', function() {
+        var hmdaJson = {};
+        var topLevelObj = {};
+
+        beforeEach(function() {
+            hmdaJson = JSON.parse(JSON.stringify(require('./testdata/complete.json')));
+            topLevelObj = hmdaJson.hmdaFile.transmittalSheet;
+            global._HMDA_JSON.hmdaFile = hmdaJson.hmdaFile;
+        });
+
+        it('should return a list of errors for a list of line counts', function(done) {
+            var handleUniqueLoanNumberErrors = rewiredEngine.__get__('handleUniqueLoanNumberErrors');
+            var counts = require('./testdata/counts.json');
+            var errors = require('./testdata/loan-number-errors.json');
+
+            expect(Object.equals(handleUniqueLoanNumberErrors(counts), errors)).to.be(true);
+            done();
+        });
+    });
+
+    Object.equals = function(x, y) {
```

Comment:
It looks like this came from SO (http://stackoverflow.com/a/6713782) in which case you should add attribution. However, why not just use [underscore's isEqual()](http://underscorejs.org/#isEqual)?